### ISSUE TITLE
io.ascii.read  no longer automatically recognizes tables in fixed_width_twoline format

### ISSUE THREAD

**Issue by kslong** [astropy/astropy#5398]
In the past, that is pre u'1.2.dev15800, one could write an asciii table in fixed_width_two_line and then read it back (allowing the routine to guess the format), but now when you do this (possibly only when the columns are strings) it is interpreting the second line in the table (the line containing ---) as values in the columns.  

A short program and the data I was using are below:
- [foo.py.txt](https://github.com/astropy/astropy/files/516784/foo.py.txt)
- [all_stat.ls.txt](https://github.com/astropy/astropy/files/516785/all_stat.ls.txt)

I had thought that fixed_width_two_line was one of the formats that ascii.read was guaranteed to recognize (which was the reason I had selected it as my default output style), and so am surprised that this feature has apparently been "lost".   As far as I remember that is how the old documentation read.

The  new [documentaion](http://docs.astropy.org/en/stable/io/ascii/index.html#guess-table-format) now appears to prefer a more basic format, but it seems like this would be an easy problem to fix, since a fixed-header is the only one that would have a line with all --

**Comment by pllim** [comment]
I remember reading some comment from @taldcroft that setting `guess=False` can fix some parsing problems. Not sure if it applies to your case, but you can give it a try. Also, by "now", which version number are you referring to?

**Comment by kslong** [comment]
I can fix the problem in my programs by setting the format when I read data in.  The issue from my perspective is that “guessing” no longer works.

Note: I included the astropy version at the beginning of my original message.  If there is another version number that is needed, please let me know

**Comment by pllim** [comment]
It is `1.3.dev...` now. Have you tried with the latest dev?

**Comment by kslong** [comment]
No, I have not tried 1.3.dev

**Comment by taldcroft** [comment]
The problem here is that your table file only has one column. One of the rules for guessing a format is that it must have at least two columns. For some formats you can get false positives if you allow an output with just one column.  I.e. Think about comma separated where you try to read with space delimited.

**Comment by kslong** [comment]
Fair enough, I guess. But then perhaps there should at least be a warning.  In practice, ascii.read does read the one-column table and the routine proceeds merrily along.

**Comment by hamogu** [comment]
This has been open for a long while and I'm not sure what to do with it.
As @taldcroft points out, we've seen a surprising number of false positives when we allow the guessing feature to succeed with just one column. As he says, certain reader will pass for any table with that (a plain CSV that just reads each row into a string is the most obvious but not the only format). That's rarely what the user wants. 
If we issue a warning for every format that comes back wit ha single column ("Reader tried format XXX but moved on because it returned only a single column") then a typical guessing process will print several warnings, even if it returns a valid multi-column table in the end. We don't want users to have to sift through several warning every time they read a table pointing out which other formats might also work, but probably didn't because everything is mushed together into a single column.

It's been like this for a decade now, and I don't want to suddenly spring up new warnings for the relatively rate use case of files that actually contain only a single column. while those of course exist, most tables that I use have two or more columns. (For example, searching a random sample of about a dozen ApJ and A&A papers I have open right now and looking at all the tables that I'm using in my current project, I don't see a single table that only has one column.) I don't say that one-column tables don't exist, or can't be useful, but it's a relatively rare case. The number of multi-column tables that would come out wrong if higher.

So, I suggest to leave the current behavior. If you have a single-column file, you can always set the format explicitly, and astropy will just read it, without second guessing if there maybe should be more columns than it finds.
